File: app/src/jvx/java/art/coded/wireframe/viewmodel/ListViewModel.java
package art.coded.wireframe.viewmodel;

import android.app.Application;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import art.coded.wireframe.model.entity.Element;
import art.coded.wireframe.model.ElementRepository;

import java.util.List;

public class ListViewModel extends ViewModel {

    static final String LOG_TAG = ListViewModel.class.getSimpleName();

    private ElementRepository mRepository;
    private LiveData<List<Element>> mData;

    public void loadData(Application application) {
        mRepository = new ElementRepository(application);
        mData = mRepository.getAllElements();
    }

    public LiveData<List<Element>> getData() {
        return mData;
    }
    public void removeData(Element element) { mRepository.delete(element); }
}

File: app/src/main/java/art/coded/wireframe/viewmodel/ListViewModel.kt
package art.coded.wireframe.viewmodel

import androidx.lifecycle.LiveData
import android.app.Application
import art.coded.wireframe.model.ElementRepository
import androidx.lifecycle.ViewModel
import art.coded.wireframe.model.entity.Element

class ListViewModel : ViewModel() {
    private var mRepository: ElementRepository? = null
    var data: LiveData<List<Element?>?>? = null
        private set

    fun loadData(application: Application?) {
        mRepository = ElementRepository(application)
        data = mRepository!!.allElements
    }

    fun removeData(element: Element?) {
        mRepository!!.delete(element)
    }

    companion object {
        val LOG_TAG = ListViewModel::class.java.simpleName
    }
}